app/views/static/retirement-form.html.haml

```diff
@@ -0,0 +1,64 @@
+%form#form_div.form-horizontal{"name"          => "angularForm",
+                               "miq-form"      => true,
+                               "model"         => "vm.retirementInfo",
+                               "model-copy"    => "vm.modelCopy",
+                               "form-changed"  => true }
+  %h3
+    = _('Set/Remove Retirement Date')
+  %tbody
+    .form-group
+      %label.col-md-3.control-label
+        = _('Enter Retirement Date as')
+      .col-md-4{"ng-init" => "formMode = 'date'"}
+        = select_tag("formMode",
+                     options_for_select(_("Specific Date and Time") => 'date', _("Time Delay from Now") => 'delay'),
+                     "ng-model"  => "formMode",
+                     "pf-select" => true)
+    .form-group{"ng-if" => "formMode == 'date'"}
+      %label.col-md-3.control-label
+        = _('Retirement Date and Time')
+      .col-md-4#retirement_date_div
+        = datetimepicker_input_tag('retirement_date_datepicker', nil,
+                                   'class'      => 'form-control',
+                                   'ng-model'   => 'vm.retirementInfo.retirementDate',
+                                   'start-date' => 'vm.datepickerStartDate')
+    %datetime-delay-picker{'ng-if'      => "formMode == 'delay'",
+                           'model'      => 'vm.retirementInfo.retirementDate',
+                           'start-date' => 'vm.datepickerStartDate'}
+    .form-group{"ng-if" => "formMode == 'delay'"}
+      %label.col-md-3.control-label
+        = _('Retirement Date and Time')
+      .col-md-4#retirement_date_result_div
+        %input.form-control#retirement_date{'readonly' => true, 'ng-model' => 'vm.retirementInfo.retirementDate'}
+    .form-group
+      %label.col-md-3.control-label
+        = _('Retirement Warning')
+      .col-md-8
+        - select_options = [{:value => "", :label => _("None")},
+                            {:value => 7,  :label => _("1 Week before retirement")},
+                            {:value => 14, :label => _("2 Weeks before retirement")},
+                            {:value => 30, :label => _("30 Days before retirement")}]
+        = select_tag("retirementWarning",
+                     "",
+                     "ng-options"  => "item.value as item.label for item in #{select_options.to_json.gsub('"', '\'')}",
```

Comment:
we should probably kill this magic...

The definition of `select_options` can live in your JS controller, and so we can just use ...`in vm.select_options` insteead of the `to_json.gsub` combo.